Based:
Description: Build the X.Org driver only for the selected system X Server version.
Author: Michael Meskes <meskes@debian.org>, Felix Geyer <debfx-pkg@fobos.de>

diff --git a/src/VBox/Additions/common/crOpenGL/Makefile.kmk b/src/VBox/Additions/common/crOpenGL/Makefile.kmk
--- a/src/VBox/Additions/common/crOpenGL/Makefile.kmk
+++ b/src/VBox/Additions/common/crOpenGL/Makefile.kmk
@@ -63,16 +63,12 @@ VBoxOGL_TEMPLATE       = VBOXCROGLR3GUESTDLL
 VBoxOGL_INCS           = .
 if1of ($(KBUILD_TARGET), linux solaris freebsd)
  VBoxOGL_INCS     += \
-	$(VBOX_PATH_X11_ROOT)/libXdamage-1.1 \
-	$(VBOX_PATH_X11_ROOT)/libXcomposite-0.4.0 \
-	$(VBOX_PATH_X11_ROOT)/libXfixes-4.0.3 \
-	$(VBOX_PATH_X11_ROOT)/damageproto-1.1.0 \
-	$(VBOX_PATH_X11_ROOT)/compositeproto-0.4 \
-	$(VBOX_PATH_X11_ROOT)/fixesproto-4.0 \
-	$(VBOX_PATH_X11_ROOT)/libx11-1.1.5-other \
-	$(VBOX_PATH_X11_ROOT)/1.3/xorg \
+	/usr/include/x11 \
+	/usr/include/xorg \
+	/usr/include/pixman-1 \
 	$(VBOX_MESA_INCS) \
-	$(PATH_ROOT)/src/VBox/Additions/x11/x11include/libdrm-2.4.13
+	/usr/include/drm \
+	/usr/include/libdrm
  VBoxOGL_DEFS     += VBOX_NO_NATIVEGL
 endif
 
diff --git a/src/VBox/Additions/x11/vboxmouse/Makefile.kmk b/src/VBox/Additions/x11/vboxmouse/Makefile.kmk
--- a/src/VBox/Additions/x11/vboxmouse/Makefile.kmk
+++ b/src/VBox/Additions/x11/vboxmouse/Makefile.kmk
@@ -53,9 +53,18 @@
 	mouse.c \
 	pnp.c \
 	VBoxUtils_68.c
-endif
-
 
+DLLS += vboxmouse_drv_17
+vboxmouse_drv_17_TEMPLATE = VBOXGUESTR3XORGMOD
+vboxmouse_drv_17_DEFS = XFree86Server IN_MODULE XFree86Module XFree86LOADER XINPUT XORG_7X IN_XF86_MODULE \
+		NO_ANSIC
+vboxmouse_drv_17_INCS := \
+   /usr/include/x11 \
+   /usr/include/xorg \
+   /usr/include/pixman-1
+vboxmouse_drv_17 = \
+	vboxmouse_15.c
+else
 #
 # vboxmouse_drv_70
 #
@@ -154,7 +166,6 @@
 	vboxmouse_15.c
 
 
-ifneq ($(KBUILD_TARGET), linux)
 
 #
 # vboxmouse_drv_17
@@ -256,6 +267,8 @@
 	$(QUIET)$(APPEND) -t "$@" "done"
      endif
 
+ifneq ($(KBUILD_TARGET), linux)
+
      TESTING += $(vboxmouse_drv_70_0_OUTDIR)/tstvboxmouse70.run
      OTHERS += $(vboxmouse_drv_70_0_OUTDIR)/tstvboxmouse70.run
 $$(vboxmouse_drv_70_0_OUTDIR)/tstvboxmouse70.run: $$(vboxmouse_drv_70_1_STAGE_TARGET)
@@ -304,8 +317,6 @@
 	    $(vboxmouse_drv_16_1_STAGE_TARGET) $(VBOXMOUSE_SRC_PATH)/undefined_15
 	$(QUIET)$(APPEND) -t "$@" "done"
 
-ifneq ($(KBUILD_TARGET), linux)
-
      TESTING += $(vboxmouse_drv_17_0_OUTDIR)/tstvboxmouse17.run
      OTHERS += $(vboxmouse_drv_17_0_OUTDIR)/tstvboxmouse17.run
 $$(vboxmouse_drv_17_0_OUTDIR)/tstvboxmouse17.run: $$(vboxmouse_drv_17_1_STAGE_TARGET)
--- a/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2011-12-19 12:58:57.000000000 +0000
+++ b/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2012-01-12 03:47:06.381765975 +0000
@@ -88,6 +88,42 @@ vboxvideo_drv_SOURCES = \
 	$(PATH_ROOT)/src/VBox/Runtime/common/alloc/heapsimple.cpp \
 	$(PATH_ROOT)/src/VBox/Runtime/common/alloc/heapoffset.cpp
 
+DLLS += vboxvideo_drv_17
+vboxvideo_drv_17_TEMPLATE = VBOXGUESTR3XORGMOD
+vboxvideo_drv_17_CFLAGS += -Wno-conversion -Wno-unused-parameter $(VBOX_GCC_Wno-variadic-macros) # template?
+vboxvideo_drv_17_DEFS := XFree86Server IN_MODULE XFree86Module XFree86LOADER XORG_7X RENDER=1 \
+	IN_XF86_MODULE IN_RT_STATIC VBOXVIDEO_13 NO_ANSIC PCIACCESS VBOX_NO_LOW_COLOUR VBOX_DRI
+vboxvideo_drv_17_INCS = \
+	/usr/include/drm \
+	/usr/include/X11/dri \
+	/usr/include/pixman-1 \
+	/usr/share/xorg-x11-server-source \
+	/usr/share/xorg-x11-server-source/fb \
+	/usr/share/xorg-x11-server-source/hw/xfree86/common \
+	/usr/share/xorg-x11-server-source/hw/xfree86/ddc \
+	/usr/share/xorg-x11-server-source/hw/xfree86/dixmods/extmod \
+	/usr/share/xorg-x11-server-source/hw/xfree86/dri \
+	/usr/share/xorg-x11-server-source/hw/xfree86/i2c \
+	/usr/share/xorg-x11-server-source/hw/xfree86/int10 \
+	/usr/share/xorg-x11-server-source/hw/xfree86/modes \
+	/usr/share/xorg-x11-server-source/hw/xfree86/os-support \
+	/usr/share/xorg-x11-server-source/hw/xfree86/os-support/bus \
+	/usr/share/xorg-x11-server-source/hw/xfree86/ramdac \
+	/usr/share/xorg-x11-server-source/hw/xfree86/shadowfb \
+	/usr/share/xorg-x11-server-source/hw/xfree86/vbe \
+	/usr/share/xorg-x11-server-source/hw/xfree86/vgahw \
+	/usr/share/xorg-x11-server-source/hw/xquartz/xpr \
+	/usr/share/xorg-x11-server-source/include \
+	/usr/share/xorg-x11-server-source/mi \
+	/usr/share/xorg-x11-server-source/miext/damage \
+	/usr/share/xorg-x11-server-source/randr \
+	/usr/share/xorg-x11-server-source/Xext \
+	/usr/share/xorg-x11-server-source/render \
+	/usr/share/xorg-x11-server-source/hw/xfree86/parser
+vboxvideo_drv_17_INCS += $(PATH_ROOT)/src/VBox/Runtime/include
+vboxvideo_drv_17_SOURCES = $(vboxvideo_drv_SOURCES) edid.c vboxvideo_dri.c
+
+ifdef NOT_FEDORA_VBOX_WITH_TESTCASES
 
 #
 # vboxvideo_drv_70
@@ -354,7 +390,6 @@ vboxvideo_drv_111_SOURCES := $(vboxvideo
 
 # Check the undefined symbols in the X.Org modules against lists of allowed
 # symbols.  Not very elegant, but it will catch problems early.
-ifdef VBOX_WITH_TESTCASES
 # ifndef VBOX_ONLY_ADDITIONS
   if1of ($(KBUILD_TARGET),linux solaris)
    ifeq ($(KBUILD_HOST_ARCH),$(KBUILD_TARGET_ARCH))
--- a/src/VBox/Additions/x11/vboxvideo/testcase/Makefile.kmk	2012-01-12 19:57:53.762613198 +0000
+++ b/src/VBox/Additions/x11/vboxvideo/testcase/Makefile.kmk	2012-01-12 20:01:17.296596446 +0000
@@ -36,11 +36,11 @@ endif # !VBOX_ONLY_SDK
 #
 tstSetModeXOrg_TEMPLATE = VBOXR3TSTEXE
 tstSetModeXOrg_CFLAGS += -std=c99
-tstSetModeXOrg_DEFS = $(filter-out IN_RT_STATIC,$(vboxvideo_drv_15_DEFS)) TESTCASE
+tstSetModeXOrg_DEFS = $(filter-out IN_RT_STATIC,$(vboxvideo_drv_17_DEFS)) TESTCASE
 tstSetModeXOrg_SOURCES  = \
 	tstSetModeXOrg.c \
 	../setmode.c
-tstSetModeXOrg_INCS = $(vboxvideo_drv_15_INCS)
+tstSetModeXOrg_INCS = $(vboxvideo_drv_17_INCS)
 
 
 # generate rules.
